client/packages/common/src/hooks/useUrlQuery/useUrlQuery.ts

```diff
@@ -4,6 +4,17 @@ interface UrlQueryObject {
   [key: string]: string | number | boolean;
 }
 
+// here you can override the parsers for specific query params
+// an example is the filter, which must allow filtering by numeric codes
+// if this is parsed as numeric, the query param changes filter=0300 to filter=300
+// which then does not match against codes, as the filter is usually a 'startsWith'
+const DefaultParsers: Record<
+  string,
+  (value: string) => string | boolean | number
+> = {
+  filter: (value: string) => value,
+};
+
```

Comment:
Ah right, yes -- we can't assume that a number string should be a number. I guess this is the downside of relying on url query as primary source of truth since everything is a string in the query.

Just wondering -- do you think it's reasonable to assume that anything we don't want converted to `Number` or `Boolean` should just remain a string?

If so, then could we simplify this further just by passing the hook a list of keys that should be ignored by the `parseSearchParams` function.

```ts
export const useUrlQuery = ( ignoreList: string[]  = []) => { 
...
```

Then `parseSearchParams` would just have one additional line:

```ts
if (ignoreList.includes(key)) return [key, value]
```

And then in `useUrlQueryParams` we'd pass the ignore list:
```ts
  const { urlQuery, updateQuery } = useUrlQuery( ["filter"] );
```

I kind of like the idea of keeping `useUrlQuery` as a purely generic wrapper for `useSearchParams` and all of the app-specific requirements being at a higher level, but that's probably just me.

Let me know what you think?

Reply:
Or the parameter could be an object so it's clearer what you're passing in `useUrlQueryParams`:
```
  const { urlQuery, updateQuery } = useUrlQuery( { ignore: ["filter"] } );
```

Reply:
that's good - thanks Carl. Quite right about keeping app specific details out of `useUrlQuery`.
Was going to use an object to make it clearer, so good thing you popped that in! 😉